packages/adminrouter/extra/src/test-harness/tests/test_common.py

```diff
@@ -0,0 +1,26 @@
+import grp
+
+import psutil
+import pytest
+
+
+# A hack to be able to parameterize a test with multiple fixtures.
+# https://github.com/pytest-dev/pytest/issues/349#issuecomment-189370273
+@pytest.fixture(
+    params=[
+        'agent_ar_process_pertest',
+        'master_ar_process_pertest'
+    ])
+def ar_process(request):
+    return request.getfuncargvalue(request.param)
+
+
+class TestNginxWorkersGroup:
+    def test_if_nginx_workers_on_master_have_correct_group(
+            self, ar_process):
+        gid = grp.getgrnam('dcos_adminrouter').gr_gid
+        for proc in psutil.process_iter():
+            if proc.pid == ar_process.pid:
+                if proc.children():
+                    for child in proc.children():
+                        assert gid in child.gids(), "Process not running as dcos_adminrouter"
```

Comment:
psutils's `Process.gids()` returns a tuple equivalent to what `os.getresgid()` would return, and that is documented with

> Return a tuple `(rgid, egid, sgid)` denoting the current process’s real, effective, and saved group ids.

Can and/or should we tighten the check? `assert gid in child.gids()` just requires that one of the three items matches `gid`.